submission/distributor_test.go

```diff
@@ -102,10 +118,75 @@ func TestNewDistributorLogClients(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			_, err := NewDistributor(tc.ll, ctpolicy.ChromeCTPolicy{}, tc.lcBuilder)
 			if gotErr, wantErr := err != nil, tc.errRegexp != nil; gotErr != wantErr {
-				t.Errorf("Expected error state does not match produced one.")
+				t.Errorf("Got error = %v, expected error = %v", (err == nil), (tc.errRegexp == nil))
 			} else if tc.errRegexp != nil && !tc.errRegexp.MatchString(err.Error()) {
 				t.Errorf("Error %q did not match expected regexp %q", err, tc.errRegexp)
 			}
 		})
 	}
 }
+
+// Mock for AddLogCLient interface
+type mockLogClient struct {
+	logURL string
+}
+
+func (m mockLogClient) AddChain(ctx context.Context, chain []ct.ASN1Cert) (*ct.SignedCertificateTimestamp, error) {
+	return nil, nil
+}
+
+func (m mockLogClient) AddPreChain(ctx context.Context, chain []ct.ASN1Cert) (*ct.SignedCertificateTimestamp, error) {
+	return nil, nil
+}
+
+func (m mockLogClient) GetAcceptedRoots(ctx context.Context) ([]ct.ASN1Cert, error) {
+	roots := []ct.ASN1Cert{}
+	if certs, ok := RootsCert[m.logURL]; ok {
+		for _, cert64 := range certs {
+			cert, err := base64.StdEncoding.DecodeString(cert64)
+			if err != nil {
+				return nil, err
+			}
+			roots = append(roots, ct.ASN1Cert{Data: cert})
+		}
+	}
+	return roots, nil
+}
+
+// buildNoLogClient is LogClientBuilder that always fails.
+func buildMockLogClient(log *loglist.Log) (client.AddLogClient, error) {
+	return mockLogClient{logURL: log.URL}, nil
+}
+
+func TestNewDistributorRootPools(t *testing.T) {
+	testCases := []struct {
+		name    string
+		ll      *loglist.LogList
+		rootNum map[string]int
+	}{
+		{
+			name:    "MockRoots",
+			ll:      sampleValidLogList(t),
+			rootNum: map[string]int{"ct.googleapis.com/aviator/": 0, "ct.googleapis.com/rocketeer/": 1, "ct.googleapis.com/icarus/": 1}, // aviator is not active
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			dist, _ := NewDistributor(tc.ll, ctpolicy.ChromeCTPolicy{}, buildMockLogClient)
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancel()
+
+			dist.Run(ctx)
+			for logURL, expectedNum := range tc.rootNum {
```

Comment:
```suggestion
			for logURL, wantNum := range tc.rootNum {
```

nit: The conventional names for expected and actual variables in Go tests are "want" and "got".

Reply:
changed